solutions/usaco-740.mdx

```diff
@@ -2,12 +2,154 @@
 id: usaco-740
 source: USACO Silver 2017 US Open
 title: Where's Bessie?
-author: Qi Wang
+author: Qi Wang, Pranav Jadhav
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_where_silver_open17.html)
 
 <LanguageSection>
+
+<CPPSection>
+```cpp
+#include <bits/stdc++.h> 
+using namespace std;
+
+const int MAX_N = 20;
+int n;
+vector<vector<char>> image(MAX_N, vector<char>(MAX_N));
+vector<vector<bool>> visited(MAX_N, vector<bool>(MAX_N));
+int PCL_count = 0;
+
+// PCL described by top-left (i1,j1) and bottom-right (i2,j2) corners
+struct PCL {
+	int i1, j1, i2, j2;
+};
+
+// Floodfill to find connected regions
+int i_min, i_max, j_min, j_max;
+void floodfill(int i , int j, char color) {
+	if(
+		i < i_min || j < j_min || i > i_max || j > j_max ||
+		visited[i][j] ||
+		image[i][j] != color
+	) {
+		return;
+	}
+
+	visited[i][j] = true;
+
+	floodfill(i + 1, j, color);
+	floodfill(i - 1, j, color);
+	floodfill(i, j + 1, color);
+	floodfill(i, j - 1, color);
+}
+
+// Check whether a given region is a PCL
+bool is_PCL(int i1, int j1, int i2, int j2) {
+
+	// keep track of num of regions for each color A-Z
+	vector<int> region_count(26, 0);
+
+	// set bounds for floodfill
+	i_min = i1;
+	i_max = i2;
+	j_min = j1;
+	j_max = j2;
+
+	// Count # of connected regions for each color present in bounds
+	for (int i = i1; i <= i2; i++) {
+		for (int j = j1; j <= j2; j++) {
+			if (!visited[i][j]) {
+				char cur_color = image[i][j];
+				region_count[cur_color - 'A']++;
+				floodfill(i, j, cur_color);
+			}
+		}
+	}
+
+	// Reset visited vector for the next call
+	fill(visited.begin(), visited.end(), vector<bool>(MAX_N, false));
+
+	// Verify PCL conditions
+	int color_count = 0;
+	bool color_with_one_region = false;
+	bool color_with_more_regions = false;
+	for (int i = 0; i < 26; i++) {
+		if (region_count[i]) {
+			color_count++;
+		}
+		if (region_count[i] == 1) {
+			color_with_one_region = true;
+		}
+		if (region_count[i] > 1) {
+			color_with_more_regions = true;
+		}
```

Comment:
```suggestion
		color_with_more_regions = region_count[i] >= 1;
```

Reply:
I am assuming you meant
```cpp
color_with_one_region = region_count[i] == 1;
color_with_more_regions = region_count[i] > 1;
```
Because otherwise `color_with_one_region` would not be accounted for.

Reply:
Oh wait actually this wouldn't work because the `color_with_one_region` and `color_with_more_regions` flags need to be true only once for any value of `i`. With an assignment like this, they might get marked true and then get marked false later on.

Reply:
hm
yeah that makes sense
then
```suggestion
		color_with_more_regions = color_with_more_regions || region_count[i] >= 1;